src/VisualStudio/Core/Test.Next/Remote/SerializationValidator.cs

```diff
@@ -94,7 +94,7 @@ public async Task<T> GetValueAsync<T>(Checksum checksum)
 
             // deserialize bits to object
             var result = Serializer.Deserialize<T>(data.Kind, reader, CancellationToken.None);
-            Contract.ThrowIfNull(result);
+            Contract.ThrowIfNull<object?>(result);
```

Comment:
Unfortunately constraining T further was tricky, this seemed like the easiest solution since this is just tests, unless somebody has a better idea.

Reply:
Going to merge with this as-is, but still open to alternatives.